pkg/agent/multicast/mcast_route.go

```diff
@@ -112,11 +114,15 @@ func (c *MRouteClient) multicastInterfacesLeaveMgroup(mgroup net.IP) error {
 // multicast route based on VIF value in the message.
 func (c *MRouteClient) processIGMPNocacheMsg(igmpMsg []byte) {
 	klog.V(2).InfoS("Received igmpMsg", "igmpMsg", igmpMsg)
-	msg, err := c.parseIGMPMsg(igmpMsg)
+	msg, warning, err := c.parseIGMPMsg(igmpMsg)
```

Comment:
why we still need this change when all messages are already discarded if it's IPAM mode? This kind of return value is not common, returning a warning in addition to an error. Normally only single error should be returned, and caller should check the type of error to decide how to proceed.

Reply:
reverted